crates/bevy_window/src/window.rs

```diff
@@ -138,6 +138,21 @@ pub struct Window {
     pub resolution: WindowResolution,
     /// Stores the title of the window.
     pub title: String,
+    /// Stores the application ID (on **`Wayland`**), `WM_CLASS` (on **`X11`**) or window class name (on **`Windows`**) of the window.
+    ///
+    /// For details about application ID conventions, see the [Desktop Entry Spec](https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#desktop-file-id).
+    /// For details about `WM_CLASS`, see the [X11 Manual Pages](https://www.x.org/releases/current/doc/man/man3/XAllocClassHint.3.xhtml).
+    /// For details about **`Windows`**'s window class names, see [About Window Classes](https://learn.microsoft.com/en-us/windows/win32/winmsg/about-window-classes).
+    ///
+    /// ## Platform-specific
+    ///
+    /// - **`Windows`**: Can only be set while building the window, setting the window's window class name.
+    /// - **`Wayland`**: Can only be set while building the window, setting the window's application ID.
+    /// - **`X11`**: Can only be set while building the window, setting the window's `WM_CLASS`.
+    /// - **`macOS`**, **`iOS`**, **`Android`**, and **`Web`**: not applicable.
+    ///
+    /// Notes: Changing this field during runtime will have no effect for now.
+    pub name: Option<String>,
```

Comment:
Could this be a &'static str instead? Given that it's not supposed to be changed?

Reply:
Sure! Done in the following commit.

Reply:
Oh it failed to build after turning the type of `name` into `Option<&'static str>`:
```rust
error: implementation of `cursor::_::_serde::Deserialize` is not general enough
   --> crates/bevy_window/src/window.rs:121:35
    |
121 | #[derive(Component, Debug, Clone, Reflect)]
    |                                   ^^^^^^^ implementation of `cursor::_::_serde::Deserialize` is not general enough
    |
    = note: `window::Window` must implement `cursor::_::_serde::Deserialize<'0>`, for any lifetime `'0`...
    = note: ...but it actually implements `cursor::_::_serde::Deserialize<'static>`
    = note: this error originates in the derive macro `Reflect` (in Nightly builds, run with -Z macro-backtrace for more info)

error: could not compile `bevy_window` (lib) due to previous error
```
How to `#[derive(Reflect)]` a type with a component whose type is `Option<&'static str>`? :thinking:

Reply:
`bevy_reflect` supports type with `'static` lifetimes, but the situation here seems more complex. Reverted it because I'm not familiar with the mechanism of `bevy_reflect`. Maybe add this optimization later.

Reply:
oh I see, no worries. It seems like `Reflect` was implemented for `&'static str` in #11686 so it's a bit strange that it still won't work.